packages/cms/pages/components/dashboard/layout/avatar-context-menu.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState } from 'react'
+import styled from 'styled-components'
+
+import { EuiIcon } from '@tensei/eui/lib/components/icon'
+import { useGeneratedHtmlId } from '@tensei/eui/lib/services/accessibility'
+import { EuiContextMenuItem } from '@tensei/eui/lib/components/context_menu/context_menu_item'
+import { EuiContextMenuPanel } from '@tensei/eui/lib/components/context_menu/context_menu_panel'
+import { EuiPopover } from '@tensei/eui/lib/components/popover'
+import { EuiAvatar } from '@tensei/eui/lib/components/avatar'
+import { EuiTextColor } from '@tensei/eui/lib/components/text'
+import { useAuthStore } from '../../../../store/auth'
+
+export const AvatarContextMenu: React.FunctionComponent = () => {
+  const { logout } = useAuthStore()
+  const [isPopoverOpen, setPopover] = useState(false)
+  const avatarButtonContextMenuPopoverId = useGeneratedHtmlId({
+    prefix: 'avatarButtonContextMenuPopover'
+  })
+
+  const onAvatarButtonClick = () => {
+    setPopover(!isPopoverOpen)
+  }
+
+  const closePopover = () => {
+    setPopover(false)
+  }
+
+  const onLogoutClick = async () => {
+    await logout()
+    window.location.href = window.Tensei.getPath('auth/login')
+    closePopover()
+  }
+
+  const items = [
+    <EuiContextMenuItem
+      key="settings"
+      icon={<EuiIcon type="gear" />}
+      onClick={closePopover}
+    >
+      Settings
+    </EuiContextMenuItem>,
+    <EuiContextMenuItem
+      key="logout"
+      icon={<EuiIcon type="exit" color="danger" />}
+      onClick={onLogoutClick}
+      color="danger"
+    >
+      <EuiTextColor color="danger">Logout</EuiTextColor>
+    </EuiContextMenuItem>
+  ]
+
+  const StyledAvatarButton = styled.button`
```

Comment:
Place your styled components outside the component. If not, they're recomputed on every render.